### PR TITLE
Removed unwanted 'stages' level from yaml

### PR DESCRIPTION
**Pull request by balandi1**
- Yaml config now starts with serverConfig level
- Made necessary changes accordingly in the template code.

### CODE REVIEW

File: code-runner.yaml [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
+      GOMODULE: on
+      GOFLAGS: -mod=vendor
+  - copy:
+      basedir: .
+      destdir: /code-runner
+  - workdir:
+      dir: /code-runner
+  - runCommand:
+      param: go build -o code-runner-server
+  - serverPort:
```
Comment (pradykaushik) [inline code-runner.yaml]: change to `port` as we already know that it's for the server.

File: configurations/dockerFileTemplate.go [hunk 2]
```diff
@@ -8,57 +8,54 @@ import (
 )
 
 type dockerfileTemplate struct {
-	Data *dockerfileData
+	Data stage
```
Comment (pradykaushik) [inline configurations/dockerFileTemplate.go]: why stage again? Didn't we get rid of this word from the config?

File: configurations/dockerYamlTemplate.go [hunk 3]
```diff
@@ -10,7 +10,7 @@ import (
 )
 
 type dockerfileDataYaml struct {
-	Stages map[string]stage `yaml:"stages"`
+	ServerConfig stage `yaml:"serverConfig"`
```
Comment (pradykaushik) [inline configurations/dockerYamlTemplate.go]: Semantics do not line up.
The member name is `ServerConfig` while the data type says that it is a `stage`. Names should be semantically correct.

**Review by pradykaushik — CHANGES_REQUESTED** [review]
Comments do not follow golang standard.
`Stage` has been included once again. Based on the discussion, it was decided that the naming would be changed.

File: code-runner.yaml (hunk 1 above)
Comment (balandi1) [reply code-runner.yaml]: Done

File: configurations/dockerFileTemplate.go (hunk 2 above)
Comment (balandi1) [reply configurations/dockerFileTemplate.go]: Yes. Renamed it

File: configurations/dockerYamlTemplate.go (hunk 3 above)
Comment (balandi1) [reply configurations/dockerYamlTemplate.go]: Yes. Renamed it

File: configurations/dockerFileTemplate.go [hunk 4]
```diff
@@ -8,57 +8,54 @@ import (
...
+func newDockerFileDataFromYamlFile(filename string) (serverConfig, error) {
 	node := yaml.Node{}
 
 	err := unmarshalYamlFile(filename, &node)
 	if err != nil {
 		return nil, fmt.Errorf("unmarshal: %v", err)
 	}
 
-	stages, err := getStagesDataFromNode(node.Content[0])
+	stageData, err := getStagesDataFromNode(node.Content[0])
```
Comment (pradykaushik) [inline configurations/dockerFileTemplate.go]: again `stage` is back!!
What is `stageData`?

File: configurations/dockerFileTemplate.go [hunk 5]
```diff
@@ -8,57 +8,54 @@ import (
...
 }
 
-// Gives all the stages and their corresponding data provided in yaml.
-func getStagesDataFromNode(node *yaml.Node) ([]stage, error) {
+// Returns the serverConfig instructions provided in yaml.
+func getStagesDataFromNode(node *yaml.Node) (serverConfig, error) {
 	var data dockerfileDataYaml
 
-	stagesInOrder, err := getStagesOrderFromYamlNode(node)
+	err := verifyStageYamlNode(node)
```
Comment (pradykaushik) [inline configurations/dockerFileTemplate.go]: what is `StageYamlNode`?

Comment (pradykaushik) [reply configurations/dockerFileTemplate.go]: The function name doesn't do much justice indicate what is being verified.

File: configurations/dockerFileTemplate.go [hunk 6]
```diff
@@ -8,57 +8,54 @@ import (
...
 		return nil, err
 	}
-	var stages []stage
-	for _, stageName := range stagesInOrder {
-		stages = append(stages, data.Stages[stageName])
-	}
-	return stages, nil
+	var stageData serverConfig
+
+	stageData = data.ServerConfig
```
Comment (pradykaushik) [inline configurations/dockerFileTemplate.go]: How is stageData the same as serverConfig?

**Review by pradykaushik — CHANGES_REQUESTED** [review]

File: configurations/dockerFileTemplate.go (hunk 4 above)
Comment (balandi1) [reply configurations/dockerFileTemplate.go]: Changed

File: configurations/dockerFileTemplate.go (hunk 5 above)
Comment (balandi1) [reply configurations/dockerFileTemplate.go]: Done

File: configurations/dockerFileTemplate.go (hunk 6 above)
Comment (balandi1) [reply configurations/dockerFileTemplate.go]: Changed

**Review by pradykaushik — APPROVED** [review]

**Pull request merged**